controllers/storageclassclaim/storageclassclaim_controller.go

```diff
@@ -92,20 +125,420 @@ func (r *StorageClassClaimReconciler) Reconcile(ctx context.Context, request rec
 	return result, nil
 }
 
+func (r *StorageClassClaimReconciler) SetupWithManager(mgr ctrl.Manager) error {
+	return ctrl.NewControllerManagedBy(mgr).
+		For(&ocsv1alpha1.StorageClassClaim{}, builder.WithPredicates(
+			predicate.GenerationChangedPredicate{},
+		)).
+		Owns(&rookCephv1.CephBlockPool{}).
+		Owns(&rookCephv1.CephFilesystemSubVolumeGroup{}).
+		Owns(&rookCephv1.CephClient{}).
+		Complete(r)
+}
+
 func (r *StorageClassClaimReconciler) reconcileConsumerPhases(ctx context.Context, storagecluster *v1.StorageCluster) (reconcile.Result, error) {
 	log := ctrllog.FromContext(ctx)
 	log.Info("Running StorageClassClaim controller in Consumer Mode")
 	return reconcile.Result{}, nil
 }
 
-func (r *StorageClassClaimReconciler) reconcileProviderPhases(ctx context.Context, storagecluster *v1.StorageCluster) (reconcile.Result, error) {
-	log := ctrllog.FromContext(ctx)
-	log.Info("Running StorageClassClaim controller in Converged/Provider Mode")
+func (r *StorageClassClaimReconciler) reconcileProviderPhases() (reconcile.Result, error) {
+	r.log.Info("Running StorageClassClaim controller in Converged/Provider Mode")
+
+	r.storageClassClaim.Status.Phase = string(v1alpha1.StorageClassClaimStateInitializing)
+
+	storageConsumerList := &ocsv1alpha1.StorageConsumerList{}
+	err := r.Client.List(r.ctx, storageConsumerList, client.InNamespace(r.namespace))
+	if err != nil {
+		return reconcile.Result{}, fmt.Errorf("failed to list storage consumer list. %v", err)
+	}
+
+	// reading storageConsumer Name from storageClassClaim ownerReferences
+	r.storageConsumer.Name = r.storageClassClaim.GetOwnerReferences()[0].Name
```

Comment:
How do you know that the correct owner is the first owner? You should look for the owner of the StorageConsumer kind.

Also if for some reason the owner list is empty the entire operator will panic on a nil ref access

Reply:
ack,